homeassistant/components/solarman/coordinator.py

```diff
@@ -0,0 +1,90 @@
+"""Coordinator for solarman integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any, cast
+
+from solarman_opendata.solarman import Solarman
+from solarman_opendata.utils import get_config
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+type SolarmanConfigEntry = ConfigEntry[SolarmanDeviceUpdateCoordinator]
+
+
+class SolarmanDeviceUpdateCoordinator(DataUpdateCoordinator):
+    """Coordinator for managing Solarman device data updates and control operations."""
+
+    config_entry: SolarmanConfigEntry
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the Solarman device coordinator."""
+        super().__init__(
+            hass,
+            logger=_LOGGER,
+            config_entry=config_entry,
+            name=DOMAIN,
+            update_interval=timedelta(
+                seconds=config_entry.data.get("scan_interval", DEFAULT_SCAN_INTERVAL)
+            ),
+        )
+
+        # Initialize the API client for communicating with the Solarman device.
+        self.api = Solarman(
+            async_get_clientsession(hass),
+            config_entry.data["host"],
+            config_entry.data["port"],
+        )
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Fetch and update device data.
+
+        This is automatically called by the DataUpdateCoordinator framework
+        according to the defined update_interval.
+        """
+        try:
+            config_data = await get_config(
+                async_get_clientsession(self.hass), self.config_entry.data["host"]
+            )
+
```

Comment:
It's interesting that we fetch the config but only use this to sync the FW version and it's not even present in the state of the coordinator